docs/programming.rst

```diff
@@ -1,50 +1,53 @@
 The PSI/J API
 =============
 
+The most important classes in this library are ``Job`` and ``JobExecutor``,
+followed by ``Launcher``.
 
-The PSI/J core
-~~~~~~~~~~~~~~
+The Job class and its modifiers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The core API contains the central components of PSI/J. The core API
-classes are independent of executor implementations. User code should
-only program against the core API. The major classes are:
+The Job-related classes listed in this section (``Job``, ``JobSpec``,
+``ResourceSpec``, and ``JobAttributes``) are independent of
+executor implementations. The authors strongly recommend that users
+program against these classes, rather than adding executor-specific
+configuration options, to the extent possible.
 
-:class:`~psij.job.Job`
-    Represents a scheduler job, including all its details and state.
+.. autoclass:: psij.Job
+    :members:
+    :noindex:
 
-:class:`~psij.job_spec.JobSpec`
-    Contains the static description of a job at a level that would be
-    supported by the POSIX interface, such as path to the executable,
-    arguments, etc.
+.. autoclass:: psij.JobStatus
+    :members:
+    :noindex:
 
-:class:`~psij.resource_spec.ResourceSpec`
+.. autoclass:: psij.JobState
+    :members:
+    :noindex:
 
-    Describes resource requirements for the job, such as number of nodes,
-    etc.
+Job modifiers
```

Comment:
Can we explain the term "modifiers" a bit? Does it refer to the spec + its properties?

Reply:
I was just looking for a very generic word to summarize what all the ``JobFoo`` classes have in common. And what I thought was, well, they all change the LRM job's behavior in one way or another.

Reply:
Right. Well, they define what the job is and what it needs, which is why we used "Specification". We could change that, but let's try to keep terms to a minimum if we can.